web/src/lib/api.ts

```diff
@@ -92,25 +88,23 @@ export default class API {
   /**
    * Ask the server for a clip
    */
-  getRandomClip(): Promise<Clip> {
-    return this.fetch('upload/random/', { responseType: 'blob', headers: {'uid': this.user.getId()}})
-      .then((req: XMLHttpRequest) => {
-        let src = window.URL.createObjectURL(req.response);
-        let glob = decodeURIComponent(req.getResponseHeader('glob'));
-        let sentence = decodeURIComponent(req.getResponseHeader('sentence'));
-        return Promise.resolve({ glob: glob, audio: src, sentence: sentence });
-      });
+  async getRandomClip(): Promise<Clip> {
+    const req = await this.fetch('upload/random/', { responseType: 'blob', headers: {'uid': this.user.getId()}})
+    let src = window.URL.createObjectURL(await req.response);
```

Comment:
Ah yeah, thanks for noticing. I must have it confused with the new Fetch API (where response & response body are promises)